### PR TITLE
Migrate to poetry

### PR DESCRIPTION
**Pull request by yalef**
- Migrate all invokes to `saritasa-invocations`
- Drop support of python 3.9
- Add support for python 3.12
- Add django 5.0 to ci matrix
- Migrate from `pip-tools` to `poetry`
- Add configuration for  `coverage` tool
- Add separate invoke for `coverage` tests run

### CODE REVIEW

File: import_export_extensions/widgets.py
Comment (NikAzanov) [inline import_export_extensions/widgets.py:290]: Why empty values shouldn't be covered?

File: pyproject.toml
Comment (NikAzanov) [inline pyproject.toml:3]: Current version is `0.4.1`, the next one will be `0.5.0` probably (or 1.0.0)

```suggestion
version = "0.4.1"
```

**Review by NikAzanov — COMMENTED** [review]
Please, check if we can get rid of `import_export_extensions/__init__.py` file content

File: .github/workflows/checks.yml
Comment (TheSuperiorStanislav) [inline .github/workflows/checks.yml:16]: Is it possible to add 3.12?

File: provision/project.py
Comment (TheSuperiorStanislav) [inline provision/project.py:7]: Let's remove this

File: pyproject.toml
Comment (TheSuperiorStanislav) [inline pyproject.toml:6]: How about adding these keywords? 
image

File: pyproject.toml
Comment (TheSuperiorStanislav) [inline pyproject.toml:1]: Please add `maintainers`, `homepage`, `repository`

File: pyproject.toml
Comment (TheSuperiorStanislav) [inline pyproject.toml:23]: No need for that, poetry can set it by itself

File: pyproject.toml
Comment (TheSuperiorStanislav) [inline pyproject.toml:37]: Let's add comments like we usually do

File: pyproject.toml
Comment (TheSuperiorStanislav) [inline pyproject.toml:27]: Please leave a comment why

File: tasks.py
Comment (TheSuperiorStanislav) [inline tasks.py:4]:
```suggestion
import provision
```

File: .github/workflows/checks.yml
Comment (yalef) [reply .github/workflows/checks.yml:16]: We're using flake8-django for linting, it doesn't support python 3.12 yet until PR doesn't merged https://github.com/rocioar/flake8-django/pull/134

File: pyproject.toml
Comment (yalef) [reply pyproject.toml:1]: We actually don't have homepage, but repository link might be useful. I'll add `documentation` also

Comment (TheSuperiorStanislav) [reply pyproject.toml:1]: What about docs page?

Comment (TheSuperiorStanislav) [reply pyproject.toml:1]: https://django-import-export-extensions.readthedocs.io/en/latest/

**Comment by yalef** [comment]
> Please, check if we can get rid of `import_export_extensions/__init__.py` file content

Only `__version__` var is used for sphinx config file, but it looks like it is optional and we can remove it

File: pyproject.toml
Comment (yalef) [reply pyproject.toml]: Hm, actually we support 3.12, but we can't run linter checks for it due issue I've described above. I'll extend version to `^3.10`

**Review by OttoAndrey — APPROVED** [review]

File: Makefile
Comment (yalef) [reply Makefile:1]: Yes, I thought about it too. Let's do it in separate PR

File: tasks.py
Comment (TheSuperiorStanislav) [inline tasks.py:17]:
```suggestion
    provision.project,
    provision.ci,
    saritasa_invocations.celery,
    saritasa_invocations.django,
    saritasa_invocations.docker,
    saritasa_invocations.pytest,
    saritasa_invocations.poetry,
    saritasa_invocations.git,
    saritasa_invocations.pre_commit,
    saritasa_invocations.mypy,
    saritasa_invocations.python,
```

File: pyproject.toml
Comment (TheSuperiorStanislav) [inline pyproject.toml:34]: Please add django 5.0

Comment (TheSuperiorStanislav) [reply pyproject.toml:34]: Or better check if poetry could set it on build. Use this to read whl files https://filext.com/file-extension/WHL
image

**Pull request merged**
